### DISCUSSION TITLE
`cargo publish` error message unhelpful with `publish = false`

### DISCUSSION THREAD

**Discussion by wooorm** [rust-lang/crates.io#5356] [Feature Requests]
```sh
$ cargo publish --dry-run
error: `xxxxxxxx` cannot be published.
The registry `crates-io` is not listed in the `publish` value in Cargo.toml.
```

It’s a bit stupid on my part, I forgot for a second that I had `publish = false` in my `Cargo.toml` to prevent publishing, but this error message threw me off.
I can imagine that some other folks would run into this too, so I’m raising this here.

How about an error message, if `publish = false`, along the lines of:

```
error: `xxxxxxxx` cannot be published.
`publish` is set to `false` in Cargo.toml and prevents publishing.
```

**Comment by carols10cents** [comment]
This error message is actually in Cargo: https://github.com/rust-lang/cargo/blob/3ff044334f0567ce1481c78603aeee7211b91623/src/cargo/ops/registry.rs#L140

Could you re-file this over in the Cargo repo please?

**Reply by wooorm** [reply]
Good point, raised here: https://github.com/rust-lang/cargo/issues/11262.
